packages/app-desktop/gui/JoplinCloudLoginScreen.tsx

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useMemo, useReducer, useState } from 'react';
+import ButtonBar from './ConfigScreen/ButtonBar';
+import { _ } from '@joplin/lib/locale';
+import { clipboard } from 'electron';
+import Button, { ButtonLevel } from './Button/Button';
+const bridge = require('@electron/remote').require('./bridge').default;
+import { uuidgen } from '@joplin/lib/uuid';
+import { Dispatch } from 'redux';
+import { reducer, intitialValues, generateLoginWithUniqueLoginCode, checkIfLoginWasSuccessful } from '@joplin/lib/services/joplinCloudUtils';
+import { AppState } from '../app.reducer';
+
+const { connect } = require('react-redux');
+
+interface Props {
+	dispatch: Dispatch;
+	joplinCloudWebsite: string;
+	joplinCloudApi: string;
+}
+
+const JoplinCloudScreenComponent = (props: Props) => {
+
+	const loginUrl = `${props.joplinCloudWebsite}/login`;
+	const applicationsUrl = `${props.joplinCloudApi}/api/applications`;
+
+	const [intervalIdentifier, setIntervalIdentifier] = useState(undefined);
+	const [state, dispatch] = useReducer(reducer, intitialValues);
+
+	const uniqueLoginCode = useMemo(() => uuidgen(), []);
+
+	const periodicallyCheckForCredentials = () => {
+		if (intervalIdentifier) return;
+
+		const interval = setInterval(async () => {
+			const response = await checkIfLoginWasSuccessful(applicationsUrl, uniqueLoginCode);
```

Comment:
If the request takes more than 2 seconds there will be multiple parallel requests and multiple dispatches. I guess it's not a problem in this particular but not ideal

Reply:
I end up doing this in a naive way:

```ts
let isWaitingResponse = false;

export const checkIfLoginWasSuccessful = async (applicationsUrl: string, ulc: string) => {
	if (isWaitingResponse) return undefined;
	isWaitingResponse = true;
...
```
I think since our interval is only 2 seconds this approach won't be a problem.